File: l33tcode/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if not digits:
            return []

        overflow = 1

        for pos in reversed(range(len(digits))):
            digits[pos] += overflow
            overflow = 0
            if digits[pos] > 9:
                digits[pos] = 0
                overflow = 1

        if overflow == 1:
            digits.insert(0, 1)

        return digits

File: l33tcode/plus-one.go
package main

func plusOne(digits []int) []int {
	remain := 1
	revResult := []int{}

	for pos := range digits {
		revPos := len(digits) - pos - 1
		newNum := digits[revPos] + remain

		revResult = append(revResult, newNum%10)
		remain = newNum / 10
	}

	if remain == 1 {
		revResult = append(revResult, remain)
	}

	result := []int{}
	for pos := range revResult {
		revPos := len(revResult) - pos - 1
		result = append(result, revResult[revPos])
	}

	return result
}
